javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -50,127 +54,472 @@ object Integer {
+      } else {
+        val buffer = new Array[Char](11)
+        var positive_value =
+          if (i < 0) -i
+          else i
+        var first_digit = 0
+        if (negative) {
+          buffer(0) = '-'
+          first_digit += 1
+        }
+
+        var last_digit  = first_digit
+        var count       = 0
+        var number: Int = 0
+        var start       = false
+        var k           = 0
+        while (k < 9) {
+          count = 0
+          number = decimalScale(k)
+          if (positive_value < number) {
+            if (start) {
+              buffer(last_digit) = '0'
+              last_digit += 1
+            }
+          }
+
+          if (k > 0) {
+            number = decimalScale(k) << 3
+            if (positive_value >= number) {
+              positive_value -= number
+              count += 8
+            }
+
+            number = decimalScale(k) << 2
+            if (positive_value >= number) {
+              positive_value -= number
+              count += 4
+            }
+          }
+
+          number = decimalScale(k) << 1
+          if (positive_value >= number) {
+            positive_value -= number
+            count += 2
+          }
+
+          if (positive_value >= decimalScale(k)) {
+            positive_value -= decimalScale(k)
+            count += 1
+          }
+
+          if (count > 0 && !start) {
+            start = true
+          }
+
+          if (start) {
+            buffer(last_digit) = (count + '0').toChar
+            last_digit += 1
+          }
+
+          k += 1
+        }
+
+        buffer(last_digit) = (positive_value + '0').toChar
+        last_digit += 1
+        count = last_digit
+        last_digit -= 1
+
+        new String(buffer, 0, count)
+      }
+    }
+  }
+
+  def toString(_i: scala.Int, _radix: scala.Int): String = {
+    if (_i == 0) {
+      "0"
+    } else {
+      val radix =
+        if (_radix < Character.MIN_RADIX || _radix > Character.MAX_RADIX) 10
+        else _radix
+      var i     = _i
+      var j     = _i
+      var count = 2
+      val negative = _i < 0
+      if (!negative) {
+        count = 1
+        j = -i
+      }
+      i /= radix
+      while (i != 0) {
+        count += 1
+        i = i / radix
+      }
+
+      val buffer = new Array[Char](count)
+      do {
+        var ch = 0 - (j % radix)
+        if (ch > 9) {
+          ch = ch - 10 + 'a'
+        } else {
+          ch += '0'
+        }
+        count -= 1
+        buffer(count) = ch.toChar
+        j /= radix
+      } while (j != 0)
+
+      if (negative) {
+        buffer(0) = '-'
+      }
+
+      new String(buffer)
+    }
+  }
 
-  @inline // because radix is almost certainly constant at call site
-  def toString(i: Int, radix: Int): String = ???
+  @inline def toUnsignedLong(x: scala.Int): scala.Long =
+    intToULong(x)
 
-  @inline def toUnsignedString(i: scala.Int): String = toUnsignedString(i, 10)
+  @inline def valueOf(i: scala.Int): Integer =
+    new Integer(i)
```

Comment:
I'm not sure whether you want to keep the complete semantics of the JDK. The JDK mandates that values from -128 to 127 be cached. (Same for `Byte`, `Short` and `Long`, IIRC).

It would probably be good for GC pressure as well, in the case SN.